### PR TITLE
[EIS] Validate EIS Gateway URL if set

### PR DESCRIPTION
**Pull request by timgrein**
This PR adds a `Setting.Validator` to validate the EIS Gateway URL set through `xpack.inference.eis.gateway.url`. If the URL is not set (for example, if you're not using EIS) we skip validation.

### CODE REVIEW

**Comment by timgrein** [comment]
@elasticmachine update branch

**Review by prwhelan — APPROVED** [review]

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/ElasticInferenceServiceSettings.java [hunk 1]
```diff
@@ -10,11 +10,57 @@
...
+     * Class to validate the EIS Gateway url set via `xpack.inference.eis.gateway.url`.
+     */
+    public static class EisGatewayURLValidator implements Setting.Validator<String> {
+
+        private static final Set<String> VALID_EIS_GATEWAY_SCHEMES = Set.of("http", "https");
+
+        @Override
+        public void validate(String value) {
+            if (Objects.isNull(value) || value.isEmpty()) {
+                // No validation needed, if eis-gateway URL is not set
```
[the remark this answers is not in the document]
Comment (timgrein) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/ElasticInferenceServiceSettings.java:44]: I've checked what gets logged on startup and it feels like it's rather uncommon to log something as specific as this (at least on the `INFO` level). There are only a few plugins logging, whether they're enabled (on a whole plugin level) like APM and security. But I think it probably doesn't cause any harm to log it on `DEBUG` level, so adding it with [Add debug log, whether eis-gateway URL is set or not](https://github.com/elastic/elasticsearch/pull/114600/commits/0457fac4c2d56868924d5efa2a56f2466bb2eba0).

**Review by vidok — APPROVED** [review]

**Comment by timgrein** [comment]
@elasticmachine update branch

**Comment by mark-vieira** [comment]
This is manifesting in serverless with the following:

```
[2024-10-15T21:34:26,251][ERROR][o.e.b.Elasticsearch      ] [es-es-index-5d88674cbc-bs9mw] fatal exception while booting Elasticsearch java.lang.IllegalArgumentException: [null] is not a valid URI scheme for the setting [xpack.inference.eis.gateway.url]. Use one of [http,https]
	at org.elasticsearch.inference@9.0.0/org.elasticsearch.xpack.inference.services.elastic.ElasticInferenceServiceSettings$EisGatewayURLValidator.validate(ElasticInferenceServiceSettings.java:60)
	at org.elasticsearch.inference@9.0.0/org.elasticsearch.xpack.inference.services.elastic.ElasticInferenceServiceSettings$EisGatewayURLValidator.validate(ElasticInferenceServiceSettings.java:37)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.common.settings.Setting.get(Setting.java:557)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.common.settings.Setting.get(Setting.java:530)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:604)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:510)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:480)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:450)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.common.settings.SettingsModule.<init>(SettingsModule.java:133)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.common.settings.SettingsModule.<init>(SettingsModule.java:51)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.node.NodeConstruction.validateSettings(NodeConstruction.java:527)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.node.NodeConstruction.prepareConstruction(NodeConstruction.java:277)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.node.Node.<init>(Node.java:200)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.bootstrap.Elasticsearch$2.<init>(Elasticsearch.java:240)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.bootstrap.Elasticsearch.initPhase3(Elasticsearch.java:240)
	at org.elasticsearch.server@9.0.0/org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:75)
```

**Comment by mark-vieira** [comment]
FYI, I've reverted this for now in https://github.com/elastic/elasticsearch/pull/114867

**Pull request merged**